### PR TITLE
Concept documentation page for fitting in Mantid

### LINKED ISSUE #15212: Add document overview of the Mantid fitting framework
Add documentation which gives a good overview of the Mantid fitting framework, includes a description of how this matches onto files the directory structure in CurveFitting, including the files in https://github.com/mantidproject/mantid/tree/master/Framework/CurveFitting/inc/MantidCurveFitting/Algorithms 

Create concept fit page

### PR DESCRIPTION
**Pull request by Anders-Markvardsen**
Description of work.

**To test:**
- Read new fitting concept page: check content, for broking links and usage example

Fixes #15212.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
@Anders-Markvardsen : is this now ready for testing? If so remember to leave the "assignee" empty.

**Comment by Anders-Markvardsen** [comment]
Thanks @FedeMPouzols

File: docs/source/concepts/Fitting.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+Consider the simple example where a model depends on just one fitting parameter, 
+here called Bob. The :math:`y_i`'s then only depend on this parameter and hence the same 
+is true for the cost function. Below is shown a made up example of how such a 
+cost function could depend on the fitting parameter Bob:
+
+.. figure:: ../images/CostFunctionVSbob.png
+   :alt: CostFunctionVSbob.png
+
+The primary problem which Mantid deals with is that of 
+local minimization, also referred to as download minimization. Referencing to
```
Comment (FedeMPouzols) [inline docs/source/concepts/Fitting.rst]: download/downhill?

File: docs/source/concepts/Fitting.rst [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+an example of sequential fitting see here: 
+http://www.mantidproject.org/MantidPlot:_Simple_Peak_Fitting_with_the_Fit_Wizard
+
+
+How do you know if a fit is successful
+--------------------------------------
+If you are fitting using the Fit Function toolbox or the Multi-dataset 
+Fitting interface then you get visual feedback as to how well the fit was. 
+
+If you are scientists experience in fitting a particular type of data
```
Comment (FedeMPouzols) [inline docs/source/concepts/Fitting.rst]: experienced?

File: docs/source/concepts/Fitting.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+The :ref:`Fit <algm-Fit>` algorithm returns an OutputStatus property,
+which can be monitored.
+
+Create an algorithm or code that aims to be clever, for a given fitting
+problem, beyond that of the generic Fit algorithm. An example of this
+is :ref:`FitPeak <algm-FitPeak>` which aims to fit single peaks 
+with some checks to ensure results are physical.
+
+If you are remain unsure if a given fit was successful then e.g. try the
```
Comment (FedeMPouzols) [inline docs/source/concepts/Fitting.rst]: are / remain

**Comment by FedeMPouzols** [comment]
It turns out that there is a concept page for FABADA: http://docs.mantidproject.org/nightly/concepts/FABADA.html. So it could be linked to when it is mentioned here.

I built the documentation and there weren't any broken links or issues (all the equations and graphs display correctly).

The documentation reads well, gives an overview of fitting in Mantid, and provides links to relevant related documentation and algorithms.

@Anders-Markvardsen : let me know if you'd like to edit the minor points I commented on. Otherwise I'd say this is ready.

A note has been added in the release notes/framework/curve fitting.

**Comment by Anders-Markvardsen** [comment]
@FedeMPouzols thanks for the read through. In my opinion the content of the concept page for FABADA should be transferred to the future FABADA minimizer page. I have added a comment to this effect to #15014

**Comment by FedeMPouzols** [comment]
:shipit:

**Comment by quantumsteve** [comment]
I agree that the test failure is unrelated and see no benefit to rerunning the system tests.

**Pull request merged**